### ISSUE TITLE
Reports: Add or review Other Info handling

### ISSUE THREAD

**Issue by kingthorin** [zaproxy/zaproxy#7717]
### Describe the bug

The following reports should have Other Info handling:
- traditional-md
- traditional-pdf
- high-level-report

The Other Info handling in Risk and Confidence HTML should be reviewed. It's currently being handled like Description.

### Steps to reproduce the behavior

Review or generate reports.

### Expected behavior

Other Info should be included as a unique field per alert.

### Software versions

Reports 0.18.0
ZAP 2.12.0 or weekly.
Any Java/OS.

### Screenshots

N/A

### Errors from the zap.log file

N/A

### Additional context

Brought to light during IRC discussion of:
- https://github.com/zaproxy/zap-extensions/pull/4381
- zaproxy/zaproxy#7260

### Would you like to help fix this issue?

- [ ] Yes

**Comment by Shershon25** [comment]
Hey! Can I take a look at this issue?

**Comment by kingthorin** [comment]
Have a look at the dev docs to get started: https://www.zaproxy.org/docs/developer/

Review the referenced issue and PR for further details.

**Comment by Avinash-Acharya** [comment]
After reviewing the referenced issue , I am assuming that bringing the other info from per alert to per instance is needed. Maybe not?

**Comment by kingthorin** [comment]
The issue is already assigned and there’s a PR being reviewed.

**Comment by kingthorin** [comment]
Disregard that comment, I’ve mixed things up.

it is assigned but there isn’t a PR yet.

**Comment by Avinash-Acharya** [comment]
I went through the PR you just mentioned. It has around 400 lines of code committed. I been trying to understand how to solve this issue as an beginning and can't find this easy to get through. 
After generating report in either way (one which has otherinfo or not), I don't find the "other info" titled report anywhere in the final report. Can you fill me in with this .

**Comment by Avinash-Acharya** [comment]
> Disregard that comment, I’ve mixed things up.
> 
> It is assigned but there isn’t a PR yet.

I just wanted to know this repo more so I was just going through the PR's and the issues. 
I had an question so thaught to ask that here. Sorry if I did something that I wasn't supposed to.

**Comment by kingthorin** [comment]
> After reviewing the referenced issue , I am assuming that bringing the other info from per alert to per instance is needed. Maybe not?

There are two issues. First some of the reports don't have other info handling and should have it. (Based on a quick review, something could have been overlooked.) Second there's one more report that's handling it similar to description, ex: that one applies to all, which isn't true some of them could be (likely are) unique per alert.

>  I had an question so thaught to ask that here. Sorry if I did something that I wasn't supposed to.

Nope you didn't, I was just trying to keep you from working on something that was already in flight. But, then I realized I was mistaken. Early land lacking ☕

**Comment by Shershon25** [comment]
Hey! Really sorry about the lag in work. I wasn't able to work much due to unexpected health problems. Looked into the issues mentioned above. I got the gist but have to dig deeper to understand the Other info part. I will submit a PR soon! (which I'm sure will be loaded with questions)

**Comment by Shershon25** [comment]
I looked into the report addon and realised a few other reports also do not have an `other info` column.  In addition to that, I wasn't able to understand how `other info` is similar to `description` in the `Risk Confidence HTML`. Even, `traditional xml` handle it in a similar fashion. Can someone elaborate on it?

**Comment by kingthorin** [comment]
You'll have to use ZAP and bit and get some data to run the reports and compare. (Or manually create alerts that have the necessary properties.  Ex: Make two with all the same properties except Other Info and see how they appear in the report when generated.)

As for some other reports also needing changes that's being handled in the PR referenced in the original issue.

**Comment by Shershon25** [comment]
Hey! I tried creating manual alerts for reports with similar properties and changed only the `Other info` status. By similar properties, you also meant the alert's name? If so, just a single instance of the alert properties are being displayed in the report along with `Other info`. Am I missing something?

**Comment by kingthorin** [comment]
Name, Risk, Confidence, Desc, etc. should be similar what should be unique per alert instance is Other Info, param, evidence, and potentially attack.

**Comment by Avinash-Acharya** [comment]
Is there any progress in this issue, If not I would like to take a look at this.

**Comment by thc202** [comment]
The traditional HTML reports were not included, was that on purpose?

**Comment by kingthorin** [comment]
Nope, I just trusted the original report/ticket. If there's something more to be done feel free to re-open and assign me.

**Comment by thc202** [comment]
That's good just wanted to make sure we weren't missing something.
